lib/finch/http2/pool.ex

```diff
@@ -161,8 +161,6 @@ defmodule Finch.HTTP2.Pool do
 
   @impl true
   def init({{scheme, host, port} = shp, registry, _pool_size, pool_opts}) do
-    {:ok, _} = Registry.register(registry, shp, __MODULE__)
```

Comment:
I remove the registering code here since we have another one after we enter connected state. IMO, registration happens after we establish the connection makes more sense